homeassistant/components/xiaomi_miio/gateway.py

```diff
@@ -33,16 +42,45 @@ def gateway_info(self):
     async def async_connect_gateway(self, host, token):
         """Connect to the Xiaomi Gateway."""
         _LOGGER.debug("Initializing with host %s (token %s...)", host, token[:5])
+
+        use_cloud = self._config_entry.options.get(CONF_CLOUD_SUBDEVICES)
+        cloud_username = self._config_entry.options.get(CONF_CLOUD_USERNAME)
+        cloud_password = self._config_entry.options.get(CONF_CLOUD_PASSWORD)
+        cloud_country = self._config_entry.options.get(CONF_CLOUD_COUNTRY)
+
         try:
             self._gateway_device = gateway.Gateway(host, token)
             # get the gateway info
             self._gateway_info = await self._hass.async_add_executor_job(
                 self._gateway_device.info
             )
+
             # get the connected sub devices
-            await self._hass.async_add_executor_job(
-                self._gateway_device.discover_devices
-            )
+            if (
+                cloud_username is not None
+                and cloud_password is not None
+                and cloud_country is not None
+                and use_cloud
+            ):
+                # use miio-cloud
+                miio_cloud = MiCloud(cloud_username, cloud_password)
```

Comment:
Should the IoT class be adjusted to `CLOUD_POLL` instead of `LOCAL_POLL` with this change? I know it can now be either/or, not sure which should be reflected in the integration as a result.

Reply:
I don't think so because even when the cloud credentials are used, the cloud is only accesed once on startup of the integration to get the subdevice list. All other communication also getting the status of the subdevices uses local communication (local polling).